Jenkinsfile

```diff
@@ -0,0 +1,269 @@
+@Library("rd-apmm-groovy-ci-library@v1.x") _
+
+/*
+ Runs the following steps in parallel and reports results to GitHub:
+ - Lint using flake8
+ - Run Python 2.7 unit tests in tox
+ - Run Python 3 unit tests in tox
+ - Build Debian packages for supported Ubuntu versions
+
+ If these steps succeed and the master branch is being built, wheels and debs are uploaded to Artifactory and the
+ R&D Debian mirrors.
+
+ Optionally you can set FORCE_PYUPLOAD to force upload to Artifactory, and FORCE_DEBUPLOAD to force Debian package
+ upload on non-master branches.
+*/
+
+pipeline {
+    agent {
+        label "16.04&&ipstudio-deps"
+    }
+    options {
+        ansiColor('xterm') // Add support for coloured output
+        buildDiscarder(logRotator(numToKeepStr: '10')) // Discard old builds
+    }
+    parameters {
+        booleanParam(name: "FORCE_PYUPLOAD", defaultValue: false, description: "Force Python artifact upload")
+        booleanParam(name: "FORCE_DEBUPLOAD", defaultValue: false, description: "Force Debian package upload")
+    }
+    environment {
+        http_proxy = "http://www-cache.rd.bbc.co.uk:8080"
+        https_proxy = "http://www-cache.rd.bbc.co.uk:8080"
+    }
+    stages {
+        stage("Clean Environment") {
+            steps {
+                sh 'git clean -dfx'
+            }
+        }
+        stage ("Tests") {
+            parallel {
+                stage ("Linting Check") {
+                    steps {
+                        script {
+                            env.lint_result = "FAILURE"
+                        }
+                        bbcGithubNotify(context: "lint/flake8", status: "PENDING")
+                        // Run the linter, excluding build directories (this can also go in the .flake8 config file)
+                        sh 'flake8 --exclude .git,.tox,dist,deb_dist,__pycache__'
```

Comment:
I assume the current consensus is to have these exclusions in .flake8?

Reply:
Probably tidier to do it there - I'll move them